packages/functional-tests/pages/signupReact.ts

```diff
@@ -1,118 +1,134 @@
   }
 
-  async setAge(value: string) {
-    const ageInput = this.getAge();
-    await ageInput.fill(value);
+  get createAccountButton() {
+    return this.page.getByRole('button', { name: 'Create account' });
   }
 
-  getAge() {
-    return this.page.getByLabel('How old are you?');
+  get codeFormHeading() {
+    return this.page.getByRole('heading', { name: /^Enter confirmation code/ });
   }
 
-  async setCode(value: string) {
-    const codeInput = this.page.getByLabel('Enter 6-digit code');
-    return codeInput.fill(value);
+  get codeTextbox() {
+    return this.page.getByTestId('confirm-signup-code-input-field');
   }
 
-  async fillOutEmailFirst(email: string) {
-    await this.setEmail(email);
-    await this.submit('Sign up or sign in');
+  get confirmButton() {
+    return this.page.getByRole('button', { name: 'Confirm' });
   }
 
-  async fillOutSignupForm(password: string, age = '21', submit = true) {
-    await this.setPassword(password);
-    await this.setPasswordConfirm(password);
-    await this.setAge(age);
-    if (submit) {
-      await this.submit('Create account');
-    }
+  get cannotCreateAccountHeading() {
+    return this.page.getByRole('heading', { name: 'Cannot create account' });
   }
 
-  async fillOutCodeForm(email: string) {
-    const code = await this.target.email.waitForEmail(
-      email,
-      EmailType.verifyShortCode,
-      EmailHeader.shortCode
+  goto(route = '/', params = new URLSearchParams()) {
+    params.set('forceExperiment', 'generalizedReactApp');
+    params.set('forceExperimentGroup', 'react');
+    return this.page.goto(
+      getReactFeatureFlagUrl(this.target, route, params.toString())
     );
-    await this.setCode(code);
-    await this.submit('Confirm');
   }
 
-  async goToEmailFirstAndCreateAccount(
-    params: URLSearchParams,
-    email: string,
-    password: string
-  ) {
-    await this.goto('/', params);
-    // fill out email first form
-    await this.fillOutEmailFirst(email);
-    await this.page.waitForURL(/signup/);
-    await this.page.waitForSelector('#root');
-    await this.fillOutSignupForm(password);
-    await this.page.waitForURL(/confirm_signup_code/);
-    await this.fillOutCodeForm(email);
-  }
+  async fillOutEmailForm(formData: EmailFormData) {
+    await expect(
+      this.emailFormHeading,
+      'Check for email form failed. The heading missing or not as expected.'
+    ).toBeVisible();
+
+    await this.emailTextbox.fill(formData.email);
 
-  async submit(label: string) {
-    await this.page.getByRole('button', { name: label }).click();
+    if (formData.submit) {
+      await this.submitButton.click();
+    }
   }
 
-  async confirmCodeHeading() {
-    this.page.getByRole('heading', { name: /Enter confirmation code/ });
+  async fillOutSignupForm(formData: SignupFormData) {
+    await expect(
+      this.signupFormHeading,
+      'Check for signup form failed. The heading missing or not as expected.'
```

Comment:
Are there guidelines on when we should have this custom error text output v not that's also in a few other tests in this PR? I think if the header is failed to be found we get a similar default error about it not being present.

Reply:
I think it's subjective. When an assertion is in a page object as opposed to the top level test method I often consider adding them

This message could probably be more clear... My thinking is that this assertion has the purpose of making sure we are in the right form before we start interacting with it and if we are not in the right form I want a more helpful message than the heading is missing. Perhaps something like:

"The signup form heading is missing or not as expected. Are you filling out the right form?"

too much https://github.com/mozilla/fxa/pull/16584/commits/653f5ab30b4dc4920945de3fc8662af41e711cc4?

Reply:
That's OK with me - I mostly ask for consistency reasons.